### ISSUE TITLE
snippets not working in Rmd

### ISSUE THREAD

**Issue by rmflight** [rstudio/rstudio#3913]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1115
    OS Version      : Linux Ubuntu 16.04
    R Version       : 3.5.0

### Steps to reproduce the problem

Open a `.Rmd` file. Attempt to execute a `snippet`, e.g. `stime`, and then `Shift+Tab`. `stime` will not convert to `Sys.time()`, as expected.

if you start a code block,

```{r}
stime
```

This will work.

Now instead do the above in the R console or a `.R` file, and it should work.

### Describe the problem in detail

Snippets don't work in .Rmd files outside of code blocks.

### Describe the behavior you expected

Previous version of RStudio (even v1.2), snippets worked in Rmd files, outside of code blocks. For example, I would use a snippet to actually insert a code block.

**Comment by kevinushey** [comment]
Is `stime` a custom snippet you've defined?

**Comment by rmflight** [comment]
Nope, first snippet defined in the list of snippets in my version of RStudio

**Comment by WolfgangRolke** [comment]
I just noticed the same thing. It started after I updated to 1.2.1114. I tried both snippets that come with RStudio and ones I wrote myself, same thing.

**Comment by jmcphers** [comment]
I was able to reproduce this issue in 1.2.1114, but not in the latest daily build (1.2.1141). @rmflight @WolfgangRolke can you confirm that it's fixed for you in the dailies?

Another possible source of confusion: R Markdown docs use two different snippet types. R snippets will work only in R code blocks. Outside code blocks, you're in Markdown mode, so only Markdown snippets will work.

**Comment by WolfgangRolke** [comment]
I actually went back to 1.1.463 and I got no problem there. I still had it under 1.2.1114. I can try 1.2.1141 at some point but not for a few days

Yes, I am talking Markdown snippets. At any rate, the same ones that didn’t work before now work again under the new (old) version.

Wolfgang

**Comment by rmflight** [comment]
Ah, I'm starting to see some confusion I introduced.

It turns out it was `markdown` snippets that were broke. I noticed when I tried to use the `r` markdown snippet to insert a code-block in an Rmd document. So it was `r` snippets were fine, and `markdown` were broke when I reported the issue. Sorry for the confusion.

However, as of 1.2.1143 (daily downloaded 2 minutes ago) all seems well. Both markdown and r snippets appear to be working.

Thanks @jmcphers !
